### PR TITLE
[CHK-2321] feat(checkout-redirect): add `CheckoutRedirectPspApiKeysConfig` bean to hold Checkout Redirect PSP API keys

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

 * add CheckoutRedirectPspApiKeysConfig bean to hold Checkout Redirect PSP API keys

#### Motivation and Context

This PR adds an object representing Checkout Redirect PSP API keys to be used by `transactions-service` and `event-scheduler-service`

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectMissingPspRequestedException.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package it.pagopa.ecommerce.commons.exceptions;
+
+import java.util.Set;
+
+/**
+ * Exception thrown when requesting an API key from NPG configuration for a
+ * nonexisting PSP
+ */
+public class CheckoutRedirectMissingPspRequestedException extends RuntimeException {
```
Comment (pietro-tota) [inline src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectMissingPspRequestedException.java:9]: same as above, this exception can cover also other configuration missing cases, such as missing logo or b.e. url

File: src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectConfigurationException.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+     * @param errorCause error cause description
+     */
+    public CheckoutRedirectConfigurationException(
+            String errorCause
+    ) {
+        super(
+                "Error parsing Checkout Redirect PSP api keys configuration, cause: %s"
+                        .formatted(errorCause)
+        );
+    }
```
[the remark this answers is not in the document]
Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectConfigurationException.java:21]: Do you mean unifying this exception with `NpgApiKeyConfigurationException` e.g. under `ConfigurationException`? (and `*MissingPspException` under `MissingConfigurationKeyException`)

[the remark this answers is not in the document]
Comment (pietro-tota) [reply src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectConfigurationException.java:21]: No, I mean that this exception can be reused for other configurations that will be used for redirections, such as missing psp url for make redirections (the business logic implementation development of this task). For what concern Npg I would leave it separate to distinguish NPG payment flow from Redirections payment ones

File: src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectMissingPspRequestedException.java (hunk 1 above)
Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectMissingPspRequestedException.java:9]: Fixed with f103dcb8

File: src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectConfigurationException.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (giovanniberti) [reply src/main/java/it/pagopa/ecommerce/commons/exceptions/CheckoutRedirectConfigurationException.java]: Fixed with f103dcb8

**Review by pietro-tota — DISMISSED** [review]

**Review by pietro-tota — APPROVED** [review]

**Pull request merged**